pkg/network/common/common_test.go

```diff
@@ -272,3 +272,56 @@ func TestParseClusterNetwork(t *testing.T) {
 		}
 	}
 }
+
+func TestParseHostSubnetEgress(t *testing.T) {
+	tests := []struct {
+		name string
+		hs   networkapi.HostSubnet
+		err  string
+	}{
+		{
+			name: "valid egress ip",
+			hs: networkapi.HostSubnet{
+				EgressIPs:   []networkapi.HostSubnetEgressIP{"10.0.0.10", "10.0.0.11"},
+				EgressCIDRs: []networkapi.HostSubnetEgressCIDR{"10.0.0.0/16"},
+			},
+			err: "",
+		},
+		{
+			name: "valid egress cidr",
+			hs: networkapi.HostSubnet{
+				EgressIPs:   []networkapi.HostSubnetEgressIP{"10.0.0.10", "10.0.0.11"},
+				EgressCIDRs: []networkapi.HostSubnetEgressCIDR{"10.0.0.0/16"},
+			},
+			err: "",
+		},
+		{
+			name: "invalid CIDR address",
+			hs: networkapi.HostSubnet{
+				EgressIPs:   []networkapi.HostSubnetEgressIP{"10.0.0.10", "10.0.0.11"},
+				EgressCIDRs: []networkapi.HostSubnetEgressCIDR{"10.139.125.80/27"},
+			},
+			err: "Invalid",
+		},
+		{
+			name: "invalid egress ip",
+			hs: networkapi.HostSubnet{
+				EgressIPs:   []networkapi.HostSubnetEgressIP{"2001:0db8:85a3:0000:0000:8a2e:0370:7334"},
+				EgressCIDRs: []networkapi.HostSubnetEgressCIDR{"10.139.125.64/27"},
+			},
+			err: "Invalid",
+		},
+	}
+	for _, test := range tests {
+		err := ValidateHostSubnetEgress(&test.hs)
+		if err == nil {
+			if len(test.err) > 0 {
+				t.Fatalf("test %q unexpectedly did not get an error", test.name)
+			}
+		} else {
```

Comment:
You have to explicitly test for the case where `test.err == "" && err != nil`, since `strings.Contains, "asdf", "") == true`

Reply:
ah good catch!